src/params.rs

```diff
@@ -72,7 +73,10 @@ impl<'a> Iterator for Params<'a> {
                 self.bound_types.clear();
                 for i in 0..self.params as usize {
                     self.bound_types.push((
-                        myc::constants::ColumnType::from(typmap[2 * i as usize]),
+                        myc::constants::ColumnType::try_from(typmap[2 * i as usize])
+                            .unwrap_or_else(|_| {
+                                panic!("got unsupported column type 0x{:x}", typmap[2 * i as usize])
+                            }),
```

Comment:
Sorry, missed this in my earlier comment — we should also propagate the `|e|` here in case it contains relevant information.

```suggestion
                            .unwrap_or_else(|e| {
                                panic!("bad column type 0x{:x}: {}", typmap[2 * i as usize], e)
                            }),
```